Mathlib/Algebra/Support.lean

```diff
@@ -217,6 +217,12 @@ theorem mulSupport_comp_eq (g : M → N) (hg : ∀ {x}, g x = 1 ↔ x = 1) (f :
 #align function.mul_support_comp_eq Function.mulSupport_comp_eq
 #align function.support_comp_eq Function.support_comp_eq
 
+@[to_additive]
+theorem mulSupport_comp_eq_of_range_subset {g : M → N} {s : Set M}
+    (hg : ∀ {x}, x ∈ s → (g x = 1 ↔ x = 1)) {f : α → M} (hf : range f ⊆ s) :
+    mulSupport (g ∘ f) = mulSupport f :=
+  Set.ext fun x ↦ not_congr <| by rw [Function.comp, hg (hf (mem_range_self x))]
```

Comment:
I don't see how `s` adds a lot here, you could always apply `hg` to `hf` yourself in the applications.
```suggestion
theorem mulSupport_comp_eq_of_range_subset {g : M → N} {f : α → M}
    (hg : ∀ {x}, x ∈ range f → (g x = 1 ↔ x = 1)) :
    mulSupport (g ∘ f) = mulSupport f :=
  Set.ext fun x ↦ not_congr <| by rw [Function.comp, hg (mem_range_self x)]
```